src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java

```diff
@@ -377,9 +377,9 @@ private void writeManifest( ConcurrentJarCreator zOut, Manifest manifest )
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         manifest.write( baos );
 
-        ByteArrayInputStream bais = new ByteArrayInputStream( baos.toByteArray() );
-        super.zipFile( createInputStreamSupplier( bais ), zOut, MANIFEST_NAME, System.currentTimeMillis(), null,
-                       DEFAULT_FILE_MODE, null, false );
+        InputStreamSupplier bais = () -> new ByteArrayInputStream( baos.toByteArray() );
```

Comment:
Unless I'm missing something in all cases where `createInputStreamSupplier`, is called it is called with byte array. Maybe we can keep the method and reduce the boilerplate even more by moving the `ByteArrayInputStream bais = new ByteArrayInputStream( ... )` part there.

Reply:
Well, I have renamed it to `in` as is the parameter name for zipFile method, there are other inconsistencies in naming, as in other places it's called payload, and source.